Towards sphinx >= 3

IAlibay opened the pull request:
Fixes #3224 #2667

Changes made in this Pull Request:
 - Unpin sphinx in CI


PR Checklist
------------
 - [x] Tests?
 - [x] Docs?
 - [x] CHANGELOG updated?
 - [x] Issue raised/referenced?

IAlibay committed:
Try unpinning sphinx

IAlibay committed:
ensure we get all the warnings for sphinx

IAlibay commented:
This is 100% duplicate object stuff... I started working on it but it's incredibly tedious work. Best thing we can do here is have folks contribute a few changes at a time until we clear all the warnings.

orbeckst committed:
[DOCS] fixed duplicate object in DensityAnalysis

orbeckst commented:
I am adding micro-commits that need to be squashed. This should make it easier to resolve issues when multiple people contribute.

orbeckst committed:
[DOC] do not index duplicated functions in encore.similarity

orbeckst committed:
[DOCS] only explicitly include hole and HoleAnalysis in docs

pep8speaks commented:
Hello @IAlibay! Thanks for updating this PR. We checked the lines you've touched for [PEP 8](https://www.python.org/dev/peps/pep-0008) issues, and found:





* In the file [`package/MDAnalysis/analysis/hole2/hole.py`](https://github.com/MDAnalysis/mdanalysis/blob/b88432416e21abdf27e86e7c3cc821292c00ee8d/package/MDAnalysis/analysis/hole2/hole.py):

> [Line 1303:80](https://github.com/MDAnalysis/mdanalysis/blob/b88432416e21abdf27e86e7c3cc821292c00ee8d/package/MDAnalysis/analysis/hole2/hole.py#L1303): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (85 > 79 characters)
> [Line 1304:80](https://github.com/MDAnalysis/mdanalysis/blob/b88432416e21abdf27e86e7c3cc821292c00ee8d/package/MDAnalysis/analysis/hole2/hole.py#L1304): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (88 > 79 characters)
> [Line 1310:80](https://github.com/MDAnalysis/mdanalysis/blob/b88432416e21abdf27e86e7c3cc821292c00ee8d/package/MDAnalysis/analysis/hole2/hole.py#L1310): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (88 > 79 characters)
> [Line 1361:80](https://github.com/MDAnalysis/mdanalysis/blob/b88432416e21abdf27e86e7c3cc821292c00ee8d/package/MDAnalysis/analysis/hole2/hole.py#L1361): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (85 > 79 characters)
> [Line 1362:80](https://github.com/MDAnalysis/mdanalysis/blob/b88432416e21abdf27e86e7c3cc821292c00ee8d/package/MDAnalysis/analysis/hole2/hole.py#L1362): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (88 > 79 characters)
> [Line 1368:80](https://github.com/MDAnalysis/mdanalysis/blob/b88432416e21abdf27e86e7c3cc821292c00ee8d/package/MDAnalysis/analysis/hole2/hole.py#L1368): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (88 > 79 characters)
> [Line 1397:80](https://github.com/MDAnalysis/mdanalysis/blob/b88432416e21abdf27e86e7c3cc821292c00ee8d/package/MDAnalysis/analysis/hole2/hole.py#L1397): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (85 > 79 characters)
> [Line 1398:80](https://github.com/MDAnalysis/mdanalysis/blob/b88432416e21abdf27e86e7c3cc821292c00ee8d/package/MDAnalysis/analysis/hole2/hole.py#L1398): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (88 > 79 characters)
> [Line 1404:80](https://github.com/MDAnalysis/mdanalysis/blob/b88432416e21abdf27e86e7c3cc821292c00ee8d/package/MDAnalysis/analysis/hole2/hole.py#L1404): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (88 > 79 characters)









* In the file [`package/MDAnalysis/core/topology.py`](https://github.com/MDAnalysis/mdanalysis/blob/b88432416e21abdf27e86e7c3cc821292c00ee8d/package/MDAnalysis/core/topology.py):

> [Line 582:1](https://github.com/MDAnalysis/mdanalysis/blob/b88432416e21abdf27e86e7c3cc821292c00ee8d/package/MDAnalysis/core/topology.py#L582): [W391](https://duckduckgo.com/?q=pep8%20W391) blank line at end of file

* In the file [`package/MDAnalysis/core/universe.py`](https://github.com/MDAnalysis/mdanalysis/blob/b88432416e21abdf27e86e7c3cc821292c00ee8d/package/MDAnalysis/core/universe.py):

> [Line 293:80](https://github.com/MDAnalysis/mdanalysis/blob/b88432416e21abdf27e86e7c3cc821292c00ee8d/package/MDAnalysis/core/universe.py#L293): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (95 > 79 characters)
> [Line 294:80](https://github.com/MDAnalysis/mdanalysis/blob/b88432416e21abdf27e86e7c3cc821292c00ee8d/package/MDAnalysis/core/universe.py#L294): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (85 > 79 characters)
> [Line 295:80](https://github.com/MDAnalysis/mdanalysis/blob/b88432416e21abdf27e86e7c3cc821292c00ee8d/package/MDAnalysis/core/universe.py#L295): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (84 > 79 characters)
> [Line 297:80](https://github.com/MDAnalysis/mdanalysis/blob/b88432416e21abdf27e86e7c3cc821292c00ee8d/package/MDAnalysis/core/universe.py#L297): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (83 > 79 characters)
> [Line 299:80](https://github.com/MDAnalysis/mdanalysis/blob/b88432416e21abdf27e86e7c3cc821292c00ee8d/package/MDAnalysis/core/universe.py#L299): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (84 > 79 characters)





##### Comment last updated at 2021-11-03 00:33:54 UTC

orbeckst commented:
This is what's currently left to do (with sphinx 4.0.2)

* [x] mdanalysis/package/MDAnalysis/analysis/pca.py:docstring of MDAnalysis.analysis.pca.PCA.transform:1: WARNING: duplicate object description of MDAnalysis.analysis.pca.PCA.transform, other instance in documentation_pages/analysis/pca, use :noindex: for one of them
* [x] mdanalysis/package/MDAnalysis/coordinates/base.py:docstring of MDAnalysis.coordinates.base.Timestep:1: WARNING: duplicate object description of MDAnalysis.coordinates.base.Timestep, other instance in documentation_pages/coordinates/FHIAIMS, use :noindex: for one of them
* [x] mdanalysis/package/MDAnalysis/coordinates/base.py:docstring of MDAnalysis.coordinates.base.Timestep:1: WARNING: duplicate object description of MDAnalysis.coordinates.base.Timestep, other instance in documentation_pages/coordinates/GRO, use :noindex: for one of them
* [x] mdanalysis/package/MDAnalysis/coordinates/base.py:docstring of MDAnalysis.coordinates.TRZ.Timestep.n_atoms:1: WARNING: duplicate object description of MDAnalysis.coordinates.TRZ.Timestep.n_atoms, other instance in documentation_pages/coordinates/TRZ, use :noindex: for one of them
* [x] mdanalysis/package/MDAnalysis/coordinates/base.py:docstring of MDAnalysis.coordinates.TRZ.Timestep.time:1: WARNING: duplicate object description of MDAnalysis.coordinates.TRZ.Timestep.time, other instance in documentation_pages/coordinates/TRZ, use :noindex: for one of them
* [x] mdanalysis/package/MDAnalysis/coordinates/TRZ.py:docstring of MDAnalysis.coordinates.TRZ.TRZReader:1: WARNING: duplicate object description of MDAnalysis.coordinates.TRZ.TRZReader, other instance in documentation_pages/coordinates/TRZ, use :noindex: for one of them
* [x] mdanalysis/package/MDAnalysis/coordinates/TRZ.py:docstring of MDAnalysis.coordinates.TRZ.TRZReader:3: WARNING: duplicate object description of MDAnalysis.coordinates.TRZ.TRZReader.ts, other instance in documentation_pages/coordinates/TRZ, use :noindex: for one of them
* [x] mdanalysis/package/MDAnalysis/coordinates/TRZ.py:docstring of MDAnalysis.coordinates.TRZ.TRZReader.Writer:1: WARNING: duplicate object description of MDAnalysis.coordinates.TRZ.TRZReader.Writer, other instance in documentation_pages/coordinates/TRZ, use :noindex: for one of them
* [x] mdanalysis/package/MDAnalysis/coordinates/TRZ.py:docstring of MDAnalysis.coordinates.TRZ.TRZReader.close:1: WARNING: duplicate object description of MDAnalysis.coordinates.TRZ.TRZReader.close, other instance in documentation_pages/coordinates/TRZ, use :noindex: for one of them
* [x] mdanalysis/package/MDAnalysis/coordinates/TRZ.py:docstring of MDAnalysis.coordinates.TRZ.TRZReader.delta:1: WARNING: duplicate object description of MDAnalysis.coordinates.TRZ.TRZReader.delta, other instance in documentation_pages/coordinates/TRZ, use :noindex: for one of them
* [x] mdanalysis/package/MDAnalysis/coordinates/TRZ.py:docstring of MDAnalysis.coordinates.TRZ.TRZReader.n_atoms:1: WARNING: duplicate object description of MDAnalysis.coordinates.TRZ.TRZReader.n_atoms, other instance in documentation_pages/coordinates/TRZ, use :noindex: for one of them
* [x] mdanalysis/package/MDAnalysis/coordinates/TRZ.py:docstring of MDAnalysis.coordinates.TRZ.TRZReader.n_frames:1: WARNING: duplicate object description of MDAnalysis.coordinates.TRZ.TRZReader.n_frames, other instance in documentation_pages/coordinates/TRZ, use :noindex: for one of them
* [x] mdanalysis/package/MDAnalysis/coordinates/TRZ.py:docstring of MDAnalysis.coordinates.TRZ.TRZReader.open_trajectory:1: WARNING: duplicate object description of MDAnalysis.coordinates.TRZ.TRZReader.open_trajectory, other instance in documentation_pages/coordinates/TRZ, use :noindex: for one of them
* [x] mdanalysis/package/MDAnalysis/coordinates/TRZ.py:docstring of MDAnalysis.coordinates.TRZ.TRZReader.skip_timestep:1: WARNING: duplicate object description of MDAnalysis.coordinates.TRZ.TRZReader.skip_timestep, other instance in documentation_pages/coordinates/TRZ, use :noindex: for one of them
* [x] mdanalysis/package/MDAnalysis/coordinates/TRZ.py:docstring of MDAnalysis.coordinates.TRZ.TRZWriter:1: WARNING: duplicate object description of MDAnalysis.coordinates.TRZ.TRZWriter, other instance in documentation_pages/coordinates/TRZ, use :noindex: for one of them
* [x] mdanalysis/package/MDAnalysis/coordinates/TRZ.py:docstring of MDAnalysis.coordinates.TRZ.TRZWriter.close:1: WARNING: duplicate object description of MDAnalysis.coordinates.TRZ.TRZWriter.close, other instance in documentation_pages/coordinates/TRZ, use :noindex: for one of them
* [x] mdanalysis/package/MDAnalysis/core/topology.py:docstring of MDAnalysis.core.topology.TransTable.atoms2residues:1: WARNING: duplicate object description of MDAnalysis.core.topology.TransTable.atoms2residues, other instance in documentation_pages/core/topology, use :noindex: for one of them
* [x] mdanalysis/package/MDAnalysis/core/topology.py:docstring of MDAnalysis.core.topology.TransTable.atoms2segments:1: WARNING: duplicate object description of MDAnalysis.core.topology.TransTable.atoms2segments, other instance in documentation_pages/core/topology, use :noindex: for one of them
* [x] mdanalysis/package/MDAnalysis/core/topology.py:docstring of MDAnalysis.core.topology.TransTable.residues2atoms_1d:1: WARNING: duplicate object description of MDAnalysis.core.topology.TransTable.residues2atoms_1d, other instance in documentation_pages/core/topology, use :noindex: for one of them
* [x] mdanalysis/package/MDAnalysis/core/topology.py:docstring of MDAnalysis.core.topology.TransTable.residues2atoms_2d:1: WARNING: duplicate object description of MDAnalysis.core.topology.TransTable.residues2atoms_2d, other instance in documentation_pages/core/topology, use :noindex: for one of them
* [x] mdanalysis/package/MDAnalysis/core/topology.py:docstring of MDAnalysis.core.topology.TransTable.residues2segments:1: WARNING: duplicate object description of MDAnalysis.core.topology.TransTable.residues2segments, other instance in documentation_pages/core/topology, use :noindex: for one of them
* [x] mdanalysis/package/MDAnalysis/core/topology.py:docstring of MDAnalysis.core.topology.TransTable.segments2atoms_1d:1: WARNING: duplicate object description of MDAnalysis.core.topology.TransTable.segments2atoms_1d, other instance in documentation_pages/core/topology, use :noindex: for one of them
* [x] mdanalysis/package/MDAnalysis/core/topology.py:docstring of MDAnalysis.core.topology.TransTable.segments2atoms_2d:1: WARNING: duplicate object description of MDAnalysis.core.topology.TransTable.segments2atoms_2d, other instance in documentation_pages/core/topology, use :noindex: for one of them
* [x] mdanalysis/package/MDAnalysis/core/topology.py:docstring of MDAnalysis.core.topology.TransTable.segments2residues_1d:1: WARNING: duplicate object description of MDAnalysis.core.topology.TransTable.segments2residues_1d, other instance in documentation_pages/core/topology, use :noindex: for one of them
* [x] mdanalysis/package/MDAnalysis/core/topology.py:docstring of MDAnalysis.core.topology.TransTable.segments2residues_2d:1: WARNING: duplicate object description of MDAnalysis.core.topology.TransTable.segments2residues_2d, other instance in documentation_pages/core/topology, use :noindex: for one of them
* [x] mdanalysis/package/MDAnalysis/core/topology.py:docstring of MDAnalysis.core.topology.TransTable.size:1: WARNING: duplicate object description of MDAnalysis.core.topology.TransTable.size, other instance in documentation_pages/core/topology, use :noindex: for one of them
* [x] mdanalysis/package/MDAnalysis/core/universe.py:docstring of MDAnalysis.core.universe.Universe.dimensions:1: WARNING: duplicate object description of MDAnalysis.core.universe.Universe.dimensions, other instance in documentation_pages/core/universe, use :noindex: for one of them
* [x] mdanalysis/package/MDAnalysis/core/universe.py:docstring of MDAnalysis.core.universe.Universe.trajectory:1: WARNING: duplicate object description of MDAnalysis.core.universe.Universe.trajectory, other instance in documentation_pages/core/universe, use :noindex: for one of them
* [x] mdanalysis/package/MDAnalysis/lib/mdamath.py:docstring of MDAnalysis.lib.mdamath.angle:1: WARNING: duplicate object description of MDAnalysis.lib.mdamath.angle, other instance in documentation_pages/lib/mdamath, use :noindex: for one of them
* [x] mdanalysis/package/MDAnalysis/lib/mdamath.py:docstring of MDAnalysis.lib.mdamath.box_volume:1: WARNING: duplicate object description of MDAnalysis.lib.mdamath.box_volume, other instance in documentation_pages/lib/mdamath, use :noindex: for one of them
* [x] mdanalysis/package/MDAnalysis/lib/mdamath.py:docstring of MDAnalysis.lib.mdamath.dihedral:1: WARNING: duplicate object description of MDAnalysis.lib.mdamath.dihedral, other instance in documentation_pages/lib/mdamath, use :noindex: for one of them
* [x] mdanalysis/package/MDAnalysis/lib/mdamath.py:docstring of MDAnalysis.lib.mdamath.norm:1: WARNING: duplicate object description of MDAnalysis.lib.mdamath.norm, other instance in documentation_pages/lib/mdamath, use :noindex: for one of them
* [x] mdanalysis/package/MDAnalysis/lib/mdamath.py:docstring of MDAnalysis.lib.mdamath.normal:1: WARNING: duplicate object description of MDAnalysis.lib.mdamath.normal, other instance in documentation_pages/lib/mdamath, use :noindex: for one of them
* [x] mdanalysis/package/MDAnalysis/lib/mdamath.py:docstring of MDAnalysis.lib.mdamath.sarrus_det:1: WARNING: duplicate object description of MDAnalysis.lib.mdamath.sarrus_det, other instance in documentation_pages/lib/mdamath, use :noindex: for one of them
* [x] mdanalysis/package/MDAnalysis/lib/mdamath.py:docstring of MDAnalysis.lib.mdamath.stp:1: WARNING: duplicate object description of MDAnalysis.lib.mdamath.stp, other instance in documentation_pages/lib/mdamath, use :noindex: for one of them
* [x] mdanalysis/package/MDAnalysis/lib/mdamath.py:docstring of MDAnalysis.lib.mdamath.triclinic_box:1: WARNING: duplicate object description of MDAnalysis.lib.mdamath.triclinic_box, other instance in documentation_pages/lib/mdamath, use :noindex: for one of them
* [x] mdanalysis/package/MDAnalysis/lib/mdamath.py:docstring of MDAnalysis.lib.mdamath.triclinic_vectors:1: WARNING: duplicate object description of MDAnalysis.lib.mdamath.triclinic_vectors, other instance in documentation_pages/lib/mdamath, use :noindex: for one of them
* [x] mdanalysis/package/MDAnalysis/selections/charmm.py:docstring of MDAnalysis.selections.charmm.SelectionWriter:1: WARNING: duplicate object description of MDAnalysis.selections.charmm.SelectionWriter, other instance in documentation_pages/selections/charmm, use :noindex: for one of them
* [x] mdanalysis/package/MDAnalysis/selections/charmm.py:docstring of MDAnalysis.selections.base.SelectionWriterBase.close:1: WARNING: duplicate object description of MDAnalysis.selections.charmm.SelectionWriter.close, other instance in documentation_pages/selections/charmm, use :noindex: for one of them
* [x] mdanalysis/package/MDAnalysis/selections/charmm.py:docstring of MDAnalysis.selections.base.SelectionWriterBase.comment:1: WARNING: duplicate object description of MDAnalysis.selections.charmm.SelectionWriter.comment, other instance in documentation_pages/selections/charmm, use :noindex: for one of them
* [x] mdanalysis/package/MDAnalysis/selections/charmm.py:docstring of MDAnalysis.selections.base.SelectionWriterBase.write:1: WARNING: duplicate object description of MDAnalysis.selections.charmm.SelectionWriter.write, other instance in documentation_pages/selections/charmm, use :noindex: for one of them
* [x] mdanalysis/package/MDAnalysis/selections/charmm.py:docstring of MDAnalysis.selections.base.SelectionWriterBase.write_preamble:1: WARNING: duplicate object description of MDAnalysis.selections.charmm.SelectionWriter.write_preamble, other instance in documentation_pages/selections/charmm, use :noindex: for one of them
* [x] mdanalysis/package/MDAnalysis/selections/gromacs.py:docstring of MDAnalysis.selections.gromacs.SelectionWriter:1: WARNING: duplicate object description of MDAnalysis.selections.gromacs.SelectionWriter, other instance in documentation_pages/selections/gromacs, use :noindex: for one of them
* [x] mdanalysis/package/MDAnalysis/selections/gromacs.py:docstring of MDAnalysis.selections.base.SelectionWriterBase.close:1: WARNING: duplicate object description of MDAnalysis.selections.gromacs.SelectionWriter.close, other instance in documentation_pages/selections/gromacs, use :noindex: for one of them
* [x] mdanalysis/package/MDAnalysis/selections/gromacs.py:docstring of MDAnalysis.selections.base.SelectionWriterBase.comment:1: WARNING: duplicate object description of MDAnalysis.selections.gromacs.SelectionWriter.comment, other instance in documentation_pages/selections/gromacs, use :noindex: for one of them
* [x] mdanalysis/package/MDAnalysis/selections/gromacs.py:docstring of MDAnalysis.selections.base.SelectionWriterBase.write:1: WARNING: duplicate object description of MDAnalysis.selections.gromacs.SelectionWriter.write, other instance in documentation_pages/selections/gromacs, use :noindex: for one of them
* [x] mdanalysis/package/MDAnalysis/selections/gromacs.py:docstring of MDAnalysis.selections.base.SelectionWriterBase.write_preamble:1: WARNING: duplicate object description of MDAnalysis.selections.gromacs.SelectionWriter.write_preamble, other instance in documentation_pages/selections/gromacs, use :noindex: for one of them
* [x] mdanalysis/package/MDAnalysis/selections/jmol.py:docstring of MDAnalysis.selections.jmol.SelectionWriter:1: WARNING: duplicate object description of MDAnalysis.selections.jmol.SelectionWriter, other instance in documentation_pages/selections/jmol, use :noindex: for one of them
* [x] mdanalysis/package/MDAnalysis/selections/jmol.py:docstring of MDAnalysis.selections.base.SelectionWriterBase.close:1: WARNING: duplicate object description of MDAnalysis.selections.jmol.SelectionWriter.close, other instance in documentation_pages/selections/jmol, use :noindex: for one of them
* [x] mdanalysis/package/MDAnalysis/selections/jmol.py:docstring of MDAnalysis.selections.base.SelectionWriterBase.comment:1: WARNING: duplicate object description of MDAnalysis.selections.jmol.SelectionWriter.comment, other instance in documentation_pages/selections/jmol, use :noindex: for one of them
* [x] mdanalysis/package/MDAnalysis/selections/jmol.py:docstring of MDAnalysis.selections.jmol.SelectionWriter.default_numterms:1: WARNING: duplicate object description of MDAnalysis.selections.jmol.SelectionWriter.default_numterms, other instance in documentation_pages/selections/jmol, use :noindex: for one of them
* [x] mdanalysis/package/MDAnalysis/selections/jmol.py:docstring of MDAnalysis.selections.base.SelectionWriterBase.write:1: WARNING: duplicate object description of MDAnalysis.selections.jmol.SelectionWriter.write, other instance in documentation_pages/selections/jmol, use :noindex: for one of them
* [x] mdanalysis/package/MDAnalysis/selections/jmol.py:docstring of MDAnalysis.selections.base.SelectionWriterBase.write_preamble:1: WARNING: duplicate object description of MDAnalysis.selections.jmol.SelectionWriter.write_preamble, other instance in documentation_pages/selections/jmol, use :noindex: for one of them
* [x] mdanalysis/package/MDAnalysis/selections/pymol.py:docstring of MDAnalysis.selections.pymol.SelectionWriter:1: WARNING: duplicate object description of MDAnalysis.selections.pymol.SelectionWriter, other instance in documentation_pages/selections/pymol, use :noindex: for one of them
* [x] mdanalysis/package/MDAnalysis/selections/pymol.py:docstring of MDAnalysis.selections.base.SelectionWriterBase.close:1: WARNING: duplicate object description of MDAnalysis.selections.pymol.SelectionWriter.close, other instance in documentation_pages/selections/pymol, use :noindex: for one of them
* [x] mdanalysis/package/MDAnalysis/selections/pymol.py:docstring of MDAnalysis.selections.base.SelectionWriterBase.comment:1: WARNING: duplicate object description of MDAnalysis.selections.pymol.SelectionWriter.comment, other instance in documentation_pages/selections/pymol, use :noindex: for one of them
* [x] mdanalysis/package/MDAnalysis/selections/pymol.py:docstring of MDAnalysis.selections.base.SelectionWriterBase.write:1: WARNING: duplicate object description of MDAnalysis.selections.pymol.SelectionWriter.write, other instance in documentation_pages/selections/pymol, use :noindex: for one of them
* [x] mdanalysis/package/MDAnalysis/selections/pymol.py:docstring of MDAnalysis.selections.base.SelectionWriterBase.write_preamble:1: WARNING: duplicate object description of MDAnalysis.selections.pymol.SelectionWriter.write_preamble, other instance in documentation_pages/selections/pymol, use :noindex: for one of them
* [x] mdanalysis/package/MDAnalysis/selections/vmd.py:docstring of MDAnalysis.selections.vmd.SelectionWriter:1: WARNING: duplicate object description of MDAnalysis.selections.vmd.SelectionWriter, other instance in documentation_pages/selections/vmd, use :noindex: for one of them
* [x] mdanalysis/package/MDAnalysis/selections/vmd.py:docstring of MDAnalysis.selections.base.SelectionWriterBase.close:1: WARNING: duplicate object description of MDAnalysis.selections.vmd.SelectionWriter.close, other instance in documentation_pages/selections/vmd, use :noindex: for one of them
* [x] mdanalysis/package/MDAnalysis/selections/vmd.py:docstring of MDAnalysis.selections.base.SelectionWriterBase.comment:1: WARNING: duplicate object description of MDAnalysis.selections.vmd.SelectionWriter.comment, other instance in documentation_pages/selections/vmd, use :noindex: for one of them
* [x] mdanalysis/package/MDAnalysis/selections/vmd.py:docstring of MDAnalysis.selections.base.SelectionWriterBase.write:1: WARNING: duplicate object description of MDAnalysis.selections.vmd.SelectionWriter.write, other instance in documentation_pages/selections/vmd, use :noindex: for one of them
* [x] mdanalysis/package/MDAnalysis/selections/vmd.py:docstring of MDAnalysis.selections.base.SelectionWriterBase.write_preamble:1: WARNING: duplicate object description of MDAnalysis.selections.vmd.SelectionWriter.write_preamble, other instance in documentation_pages/selections/vmd, use :noindex: for one of them

richardjgowers committed:
fix duplicate pca transform doc

richardjgowers committed:
remove some duplicate Timestep docs

richardjgowers committed:
duplicated transtable docs

orbeckst committed:
[DOCS] fix TRZ sphinx markup

- only explicitly include TRZReader and TRZWriter
- explicitly reference base.Timestep as type

orbeckst committed:
[DOCS] define TransTable.size as explicit doc

Hide the property doc with :meta private:, see
https://www.sphinx-doc.org/en/master/usage/extensions/autodoc.html#directives
available since sphinx 3.0

orbeckst committed:
[DOCS] update Universe documentation

- fix sphinx duplicate indexing complaints by using the explicit attribute
  docs and using :meta private: for property docs
- update return types for methods
- reorganized main Universe doc string to conform to numpydoc

orbeckst commented:
Note that one common problem is that we document a managed attribute (`property`) both in the main doc string under *Attributes* and in the doc string of the property. I opted to 
- retain the explicit doc string (and expand if necessary, using information from the property)
- make the property doc string private using an [autodoc hack](https://www.sphinx-doc.org/en/master/usage/extensions/autodoc.html#directives) by embedding `:meta private:` in the property docs [info fields list](https://www.sphinx-doc.org/en/master/usage/restructuredtext/domains.html#info-field-lists). This seems to be the only way to exclude an autodocced docstring.

I chose the explicit attr description because these tend to be important attributes that should be listed together instead of spread across. By keeping the explicit docs, the interactive doc string is also more readable.

orbeckst committed:
[DOCS] fixed duplicate indexing for selection exporters

orbeckst committed:
add CHANGELOG entry

orbeckst committed:
[DOCS] fixed duplicate indexing in lib.mdamath

orbeckst committed:
Merge branch 'develop' into sphinx-3

orbeckst commented:
Locally with sphinx 4.2 I have no more warnings and the build succeeds without erroring.

lilyminium commented on package/MDAnalysis/core/universe.py:
```diff
@@ -385,33 +407,39 @@ def empty(cls, n_atoms, n_residues=1, n_segments=1,
-          allowing coordinates to be set and written.  Default is False
+          if ``True``, attaches a
+          :class:`MDAnalysis.coordinates.memory.MemoryReader` allowing
+          coordinates to be set and written.  Default is ``False``
```
```suggestion
          coordinates to be set and written.
```

Let's start removing default values from docstrings?

lilyminium commented on package/MDAnalysis/core/universe.py:
```diff
@@ -629,22 +659,34 @@ def select_atoms(self, *args, **kwargs):
-        """Angles between atoms"""
+        """Angles between atoms.
+
+        :meta private:
```
I wonder if, if the `angles` attribute was removed in the class docstring, would this one be able to be used and not private?

lilyminium replied:
This is already answered in https://github.com/MDAnalysis/mdanalysis/pull/3438#issuecomment-958421622, sorry

orbeckst replied:
I don't think so.

orbeckst replied:
... I mean, as long as it still contains ":meta private:". If you remove the one in the class doc string then this one can be used and it will be auto-documented if it does _not_ contain ":meta private:".

lilyminium commented on package/MDAnalysis/core/universe.py:
```diff
@@ -1321,19 +1369,25 @@ def from_smiles(cls, smiles, sanitize=True, addHs=True,
-            function. Requires adding hydrogens with the `addHs` parameter
+            Generate 3D coordinates using RDKit's
+            :func:`AllChem.EmbedMultipleConfs` function. Requires adding
+            hydrogens with the `addHs` parameter
```
```suggestion
            hydrogens with the ``addHs`` parameter
```

orbeckst replied:
We actually refer to args with the default role, as far as I know.

lilyminium commented on package/MDAnalysis/core/universe.py:
```diff
@@ -1321,19 +1369,25 @@ def from_smiles(cls, smiles, sanitize=True, addHs=True,
         numConfs : int (optional, default 1)
             Number of frames to generate coordinates for. Ignored if
-            `generate_coordinates=False`
+            `generate_coordinates` is ``False``
```
```suggestion
            ``generate_coordinates=False``
```

I tend to find this clearer.

lilyminium commented on package/MDAnalysis/core/universe.py:
```diff
@@ -1321,19 +1369,25 @@ def from_smiles(cls, smiles, sanitize=True, addHs=True,
+            `generate_coordinates` is ``False``
         rdkit_kwargs : dict (optional)
-            Other arguments passed to the RDKit `EmbedMultipleConfs` function
+            Other arguments passed to the RDKit :func:`EmbedMultipleConfs`
```
This probably won't render unless it's a full path:

:func:`rdkit.Chem.rdDistGeom.EmbedMultipleConfs`

lilyminium commented on package/MDAnalysis/core/universe.py:
```diff
@@ -1321,19 +1369,25 @@ def from_smiles(cls, smiles, sanitize=True, addHs=True,
-            Generate 3D coordinates using RDKit's `AllChem.EmbedMultipleConfs`
-            function. Requires adding hydrogens with the `addHs` parameter
+            Generate 3D coordinates using RDKit's
+            :func:`AllChem.EmbedMultipleConfs` function. Requires adding
```
This probably won't render unless it's a full path:

:func:`rdkit.Chem.rdDistGeom.EmbedMultipleConfs`

orbeckst replied:
It renders as a func but it won't link — I didn't check if RDKit has intersphinx mappings.

lilyminium reviewed (approved):
Thanks for doing the work to fix the errors, @IAlibay @richardjgowers @orbeckst. Docs are building (https://mdanalysis--3438.org.readthedocs.build/en/3438/documentation_pages/core/universe.html#MDAnalysis.core.universe.Universe.angles) and look good so I'll approve -- just curious about the use of `meta private` and whether it's needed.

orbeckst commented on .github/workflows/gh-ci.yaml:
```diff
@@ -227,7 +227,7 @@ jobs:
     - name: build_docs
       run: |
-        cd package && python setup.py build_sphinx -E
+        cd package && python setup.py build_sphinx -E --keep-going
```
Are we keeping "--keep-going"?

richardjgowers replied:
I think this is what gave all warnings rather than just the first, which was very useful, so yes?

IAlibay replied:
Agreed, it's no cost to us but will be super useful in future reviews.

orbeckst committed:
Apply suggestions from code review

Co-authored-by: Lily Wang <31115101+lilyminium@users.noreply.github.com>

richardjgowers reviewed (approved):
Woo, we defeated the sphinx!
